Adding transitions; Fixing dark mode header bar; Fixing sidebar border staying while menu is closed

dandalton1 opened the pull request:
- Added transitions to sidebar list items (such as settings selectors).
- Added transitions to swap colors over when a theme is changed.
- Menu icon now animates the background shadow's opacity.
- Header bar now matches the rest of the dark theme.
- Sidebar border is no longer visible after closing menu. (Fixes #460)

dandalton1 commented:
I signed it

dandalton1 commented:
signed it again

BugsNash commented on css/theme-dark.css:
```diff
@@ -1,5 +1,15 @@
+}
+
+body[theme="dark"] #search-counting.nomatches {
+  color: #ff0000;
```
why has this changed?

dandalton1 replied:
I changed that because the red that went with that just didn't look right. Should I change it back?

BugsNash replied:
I think we should leave it. I'm wary to make color changes that haven't been approved by a designer/UX review.

BugsNash commented on css/theme-dark.css:
```diff
@@ -1,5 +1,15 @@
+
+body[theme="dark"] #search-input,
+body[theme="dark"] .mdc-icon-button {
+  color: white;
```
this should probably be var(--dark-accent-color)

dandalton1 replied:
Got'cha, will fix.

BugsNash replied:
addressed in main thread conversation about overall colors

dandalton1 commented:
![image](https://user-images.githubusercontent.com/33505454/56557946-fef63d80-656a-11e9-8a77-c15c13d793a3.png)

I just got a better idea of what you were talking about. The color change was to make the search results stand out from the darker colored search box (my eyes can read a bright color on dark way easier than they can read `#B00`, but that's my opinion).

The header color was the highlight color; I just thought it odd to make the header the same black the text editor is (that specific shade of black). I can change that to be the same as `--dark-background-color`.

Finally - I didn't use `var(--dark-accent-color)` since `--dark-accent-color` has an opacity of just `0.12`.

Due to the way that blends with the black behind it, this gives it the real rgb values of `rgb(58,58,58)`, or hex  `#3a3a3a`. That's not feasible to try and read off of an already near-black search bar.

dandalton1 commented:
![image](https://user-images.githubusercontent.com/33505454/56558613-48478c80-656d-11e9-8f23-5ab8b53b5425.png)

Here's an image of what that render should look like now; I fixed the window title bar to be that same black the background uses, I reverted my change of the red, and I desire to keep my change of the dark bar's icons to white (or, at least, somewhere close to it).

dandalton1 commented:
Fixed also Issue #460.

BugsNash commented:
Ah sorry I wasn't noting the difference between text and the outlines.

Below are some snippets of the designer's mocks for the UI update of this app. These have the same black color for the header and the rest of the app. They also have the same accent color for the outline color for the find box as the rest of the app (what i've put in the css as --dark-accent-color). The icons seem to be a little brighter, at about #909090. The text in the find box should be the same color as the sidebar text, at about #bdc1c6 (note this isn't the current value of the sidebar text color).

Also, there should still be a visible divider between the header and the rest of the app for the dark theme (as there is with the light theme), in the --dark-divider-color.

![Screen Shot 2019-04-24 at 12 56 39 PM](https://user-images.githubusercontent.com/6046079/56629372-7d003480-6690-11e9-99b0-8f6f98341fda.png)

![Screen Shot 2019-04-24 at 12 56 48 PM](https://user-images.githubusercontent.com/6046079/56629377-7f628e80-6690-11e9-8753-bf0e85525c2a.png)

BugsNash commented on js/controllers/window.js:
```diff
@@ -52,9 +52,11 @@ WindowController.prototype.initUI_ = function() {
 WindowController.prototype.initSidebar_ = function() {
   if (this.settings_.get('sidebaropen')) {
     $('#sidebar').css('width', this.settings_.get('sidebarwidth') + 'px');
+    $('#sidebar').css('border-right-width', '2px');
```
good one! this (and some of the other style setting code in this method) should probably be handled in css rather than here. can you please add a comment at the start of this method:

// FIXME: move style setting to css where possible

dandalton1 replied:
I could; although, some of the sidebar code is handled through the app's settings. For example, the state of the sidebar and its width.

The closing sidebar is done by setting its width to 0px, and opening it is done by setting it back to `n`px, where `n` is the previous width that was stored in the app settings.

Due to the nature of the way the sidebar opens, adding a class (say, `.sidebar-open`) might not be the correct way to go about it, since its width is variable.

Regardless, I'll add the comment.

BugsNash replied:
yeah some of it won't be able to be moved, which is why i said 'where possible'.

BugsNash commented on css/theme-dark.css:
```diff
@@ -1,10 +1,14 @@
 body[theme="dark"] header,
 body[theme="dark"] .search-container {
-  background-color: var(--dark-background-color)
+  background-color: var(--dark-background-color);
```
good spot with the missing semi colon, i missed that

dandalton1 commented:
From my last commit:

I made the header text the same as the already established color of the Settings sidebar text; that is, `#EBEBEB` instead of a pure `white`.

From `app.css`:

> ```css
> #sidebar {
>   ...
>   color: #EBEBEB;
>   ...
> }
> ```

I also removed an extra semi-colon and put in a semi-colon in a spot where one was missing.

BugsNash commented:
The menu bar in dark theme is looking great! 2 more things to finish it off:

1)The ripple animation on the icons on the dark background is barely visible. You can fix this by changing the :before and :after for the icons to have a background-color of white (instead of the default black).

2) the color of the lines between the previous/next icons in the find bar should match the outline of the find bar.

dandalton1 commented:
Thank you! Will commit/fix later on tonight

dandalton1 commented:
Fixed that ripple animation and those border colors.

BugsNash commented:
Perfect thank you! I'll push this to the web store soon with some new translations when they are finished.

BugsNash commented:
Just uploaded to canary channel. Will upload to stable tomorrow all going well :)

BugsNash commented:
Have uploaded to stable.
